bootstrap/commandline/common.go

```diff
@@ -0,0 +1,133 @@
+}
+
+// Parse parses the passed in command-lie arguments looking to the default set of common flags
+func (f *DefaultCommonFlags) Parse(arguments []string) {
+	// The flags package doesn't allow for String flags to be specified without a value, so to support
+	// -cp/-configProvider without value to indicate using default host value we must detect use of this option with
+	// out value and insert the default value before parsing the command line options.
+	for index, option := range arguments {
+		if strings.Contains(option, "-cp") || strings.Contains(option, "-configProvider") {
+			if !strings.Contains(option, "=") {
+				arguments[index] = "-cp=" + DefaultConfigProvider
+			}
+		}
+	}
+
+	// Usage is provided by caller, so leaving individual usage blank here so not confusing where if comes from.
+	f.FlagSet.StringVar(&f.configProviderUrl, "configProvider", "", "")
+	f.FlagSet.StringVar(&f.configProviderUrl, "cp", "", "")
+	f.FlagSet.BoolVar(&f.useRegistry, "registry", false, "")
+	f.FlagSet.BoolVar(&f.useRegistry, "r", false, "")
+	f.FlagSet.StringVar(&f.profile, "profile", "", "")
+	f.FlagSet.StringVar(&f.profile, "p", "", ".")
+	f.FlagSet.StringVar(&f.configDir, "confdir", "", "")
+
+	f.FlagSet.Usage = f.helpCallback
+
+	err := f.FlagSet.Parse(arguments)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(0)
+	}
+}
+
+// UseRegistry returns whether the Registry should be used or not
+func (f *DefaultCommonFlags) UseRegistry() bool {
+	return f.useRegistry
+}
+
+// ConfigProviderUrl returns the url for the Configuration Provider, if one was specified.
+func (f *DefaultCommonFlags) ConfigProviderUrl() string {
+	return f.configProviderUrl
+}
+
+// Profile returns the profile name to use, if one was specified
+func (f *DefaultCommonFlags) Profile() string {
+	return f.profile
+}
+
+// ConfigDirectory returns the directory where the config file(s) are located, if it was specified.
+func (f *DefaultCommonFlags) ConfigDirectory() string {
+	return f.configDir
+}
+
+// Help displays the usage help message and exit.
+func (f *DefaultCommonFlags) Help() {
+	f.helpCallback()
+}
+
+// commonHelpCallback displays the help usage message and exits
+func (f *DefaultCommonFlags) helpCallback() {
+	var defaultUsageStr = `
+Usage: %s [options]
+Server Options:
+    -cp, --configProvider           Indicates to use Configuration Provider service at specified URL.
+                                    URL Format: {type}.{protocol}://{host}:{port} ex: consul.http://localhost:8500
+    -r, --registry                  Indicates service should use Registry
+    -p, --profile <name>            Indicate configuration profile other than default
+    --confdir                       Specify local configuration directory
+`
+	var helpUsage = `
+Common Options:
+    -h, --help                      Show this message
+`
+	usage := defaultUsageStr
+	if f.additionalUsage != "" {
+		usage += f.additionalUsage
+	}
+	usage += helpUsage
+
+	fmt.Printf(usage, os.Args[0])
```

Comment:
This would be cleaner, IMO, if you reduced it down to a single fmt.Printf(...) statement.

Reply:
This is what I had in mind (to eliminate the off-putting multi-line string definition):

``` go
	fmt.Printf(
		"Usage: %s [options]\n",
		"Server Options:\n",
		"    -cp, --configProvider           Indicates to use Configuration Provider service at specified URL.\n",
		"                                    URL Format: {type}.{protocol}://{host}:{port} ex: consul.http://localhost:8500\n",
		"    -f, --file <name>               Indicates name of the local configuration file. Defaults to configuration.toml\n",
		"    -p, --profile <name>            Indicate configuration profile other than default\n",
		"    --confdir                       Specify local configuration directory\n",
		"    -r, --registry                  Indicates service should use Registry\n",
		"%s\n",
		"Common Options\n:",
		"    -h, --help                      Show this message",
		"\n",
		os.Args[0],
		f.additionalUsage,
	)
```

Reply:
Ahh, yea this also fixes the issue with IDE editor replacing spaces with tabs. Will change.

Reply:
Actual, that syntax doesn't work. Only the first string can have parameters... So it all has to be one long string.

Reply:
My mistake -- 

```
	fmt.Printf(
		"Usage: %s [options]\n"+
			"Server Options:\n"+
			"    -cp + --configProvider           Indicates to use Configuration Provider service at specified URL.\n"+
			"                                    URL Format: {type}.{protocol}://{host}:{port} ex: consul.http://localhost:8500\n"+
			"    -f + --file <name>               Indicates name of the local configuration file. Defaults to configuration.toml\n"+
			"    -p + --profile <name>            Indicate configuration profile other than default\n"+
			"    --confdir                       Specify local configuration directory\n"+
			"    -r + --registry                  Indicates service should use Registry\n"+
			"%s\n"+
			"Common Options\n:"+
			"	 -h + --help                      Show this message"+
			"\n",
		os.Args[0],
		f.additionalUsage,
	)
```

@lenny-intel